dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -0,0 +1,254 @@
+      oauth_token_expiration: 'test-value',
+      sharing_disabled: 'test-value',
+      next_census_display: 'test-value'
+    refute_equal({}, user.properties)
+
+    purge_user user
+
+    user.reload
+    assert_equal({}, user.properties)
+  end
+
+  test 'leaves urm and races for US users' do
+    user = create :student, :within_united_states, races: 'white,hispanic'
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+
+    purge_user user
+
+    user.reload
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+  end
+
+  test 'clears urm and races for non-US users' do
+    user = create :student, :outside_united_states, races: 'white,hispanic'
+    assert_equal true, user.urm
+    assert_equal 'white,hispanic', user.races
+
+    purge_user user
+
+    user.reload
+    assert_nil user.urm
+    assert_nil user.races
+  end
+
+  test 'removes StudioPerson if it was only tied to this user' do
+    user = create :teacher
+    studio_person = user.studio_person
+    refute_nil user.studio_person_id
+
+    purge_user user
+
+    user.reload
+    assert_nil user.studio_person_id
+    assert_empty StudioPerson.where(id: studio_person.id)
+  end
+
+  test "only removes association with StudioPerson if it's tied to other users" do
+    user = create :teacher
+    studio_person = user.studio_person
+    refute_nil user.studio_person_id
+    other_user = create :teacher, studio_person_id: studio_person.id
+    assert_equal user.studio_person_id, other_user.studio_person_id
+
+    purge_user user
+
+    user.reload
+    other_user.reload
+    assert_nil user.studio_person_id
+    refute_nil other_user.studio_person_id
+    refute_empty StudioPerson.where(id: studio_person.id)
+  end
+
+  test 'clears sensitive user location data' do
+    user = create :student, :within_united_states
+    user_geo = user.user_geos.first
+    refute_nil user_geo.ip_address
+    refute_nil user_geo.city
+    refute_nil user_geo.state
+    refute_nil user_geo.country
+    refute_nil user_geo.postal_code
+    refute_nil user_geo.latitude
+    refute_nil user_geo.longitude
+
+    purge_user user
+
+    user_geo.reload
+    assert_nil user_geo.ip_address
+    assert_nil user_geo.city
+    assert_nil user_geo.postal_code
+    assert_nil user_geo.latitude
+    assert_nil user_geo.longitude
+    # Note: Does not delete state or country
+    refute_nil user_geo.state
+    refute_nil user_geo.country
+  end
+
+  test 'purged student still passes validations' do
+    user = create :student
+    assert user.valid?
+
+    purge_user user
+
+    user.reload
+    assert user.valid?
+  end
+
+  test 'purged teacher still passes validations' do
+    user = create :teacher
+    assert user.valid?
+
+    purge_user user
+
+    user.reload
+    assert user.valid?
+  end
+
+  private
+
+  def purge_user(user)
+    SolrHelper.stubs(:delete_document).once
+    DeleteAccountsHelper.new(solr: {}).purge_user(user)
+  end
```

Comment:
Since it appears that we call `user.reload` after every call to `purge_user` they could be combined.

Reply:
I'm in favor of something like this, but wasn't sure if it would work like I expected (re: Ruby scopes) or whether it would read well.

```

Reply:
_Discussed in person, since this helper is private to this file and `purge_user student` pretty obviously mutates the student when read as English, deduplicating should take priority over concerns about mutating arguments._